cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -172,102 +181,103 @@ def supplemental_schema_validation(ua_config: dict):
-                parsed_url = urlparse(value)
-                if parsed_url.scheme not in ("http", "https"):
-                    errors.append(
-                        f"Expected URL scheme http/https for ua:config:{key}."
-                        f" Found: {value}"
-                    )
-            except (AttributeError, ValueError):
+        if key not in UA_CONFIG_PROPS:
+            LOG.warning(
+                "Ignoring unknown ubuntu_advantage.config.%s property", key
+            )
+            continue
+        elif value is None:  # key will be unset
+            continue
+        try:
+            parsed_url = urlparse(value)
+            if parsed_url.scheme not in ("http", "https"):
                 errors.append(
-                    f"Expected a URL for ua:config:{key}. Found: {value}"
+                    f"Expected URL scheme http/https for ua:config:{key}"
                 )
+        except (AttributeError, ValueError):
+            errors.append(f"Expected a URL for ua:config:{key}")
 
     if errors:
         raise ValueError(
-            f"Invalid ubuntu_advantage configuration:{nl}{nl.join(errors)}"
+            "Invalid ubuntu_advantage configuration:\n{}".format(
+                "\n".join(errors)
+            )
         )
 
 
-def configure_ua(token=None, enable=None, config=None):
-    """Call ua commandline client to attach or enable services."""
-    error = None
-    if not token:
-        error = "ubuntu_advantage: token must be provided"
-        LOG.error(error)
-        raise RuntimeError(error)
-
-    if enable is None:
-        enable = []
-    elif isinstance(enable, str):
-        LOG.warning(
-            "ubuntu_advantage: enable should be a list, not"
-            " a string; treating as a single enable"
-        )
-        enable = [enable]
-    elif not isinstance(enable, list):
-        LOG.warning(
-            "ubuntu_advantage: enable should be a list, not"
-            " a %s; skipping enabling services",
-            type(enable).__name__,
-        )
-        enable = []
-
-    if config is None:
-        config = dict()
-    elif not isinstance(config, dict):
-        LOG.warning(
-            "ubuntu_advantage: config should be a dict, not"
-            " a %s; skipping enabling config parameters",
-            type(config).__name__,
+def set_ua_config(ua_config: Any = None):
+    if ua_config is None:
+        return
+    if not isinstance(ua_config, dict):
+        raise RuntimeError(
+            f"ubuntu_advantage: config should be a dict, not"
+            f" a {type(ua_config).__name__};"
+            " skipping enabling config parameters"
         )
-        config = dict()
+    supplemental_schema_validation(ua_config)
 
     enable_errors = []
-
-    # UA Config
-    for key, value in sorted(config.items()):
+    for key, value in sorted(ua_config.items()):
+        if key not in UA_CONFIG_PROPS:
+            continue  # Already logged in `supplemental_schema_validation`
```

Comment:
I don't think we want to be so strict here (per my point above) there are some config settings already supported by UA that we might want to allow. This initial validation we were doing was to perform additional level of validation of known URL-type *proxy ua config values. If we want we can add the *_timer config options to cloud-init's schema, and assert that they are integers. But I still think we want to add an `additionalProperties: true`  under `ubuntu_advantage: config` in our schema-cloud-config-v1.json.